File: 2358-number-of-ways-to-split-array/2358-number-of-ways-to-split-array.java
class Solution {
    public int waysToSplitArray(int[] nums) {
        long lsum = 0, rsum = 0;
        int n = nums.length;
        for (int x : nums)
            rsum += x;

        int c = 0;
        for (int u = 0; u < n - 1; u++) {
            lsum += nums[u];
            rsum -= nums[u];
            if (lsum >= rsum)
                c++;
        }
        return c;
    }
}

File: 2358-number-of-ways-to-split-array/2358-number-of-ways-to-split-array.py
class Solution(object):
    def waysToSplitArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)
        rsum, lsum = sum(nums), 0
        cnt = 0

        for i in range(n - 1):
            lsum += nums[i]
            rsum -= nums[i]
            if lsum >= rsum:
                cnt += 1
            
        return cnt
        
